### PR TITLE
4.x Adding github CI ACTION coverage tests for Integration/Unit tests

### PR DESCRIPTION
**Pull request by luu-alex**
## Description
#5705 

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../blob/1.x/CONTRIBUTIONS.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: templates/jest.config.js.tmpl [hunk 1]
```diff
@@ -30,6 +30,13 @@ module.exports = {
 	collectCoverage: false,
 	coverageReporters: ['json'],
 	coverageDirectory: '.coverage',
+	coverageThreshold: {
+		global: {
+		  branches: 80,
+		  functions: 80,
+		  lines: 80,
```
Comment (jdevcs) [inline templates/jest.config.js.tmpl]: For now should we only check `%line` coverage >= 80% ( as we were working on it in last few sprints ), and remove/comment other checks. Once we have high coverage across all these in near future we can uncomment.

Comment (luu-alex) [reply templates/jest.config.js.tmpl]: thanks, updated.

**Comment by luu-alex** [comment]
@jdevcs  will also need to ignore failing coverage test cases until we cover lines >  80%.

**Review by nikoulai — APPROVED** [review]

**Comment by Spacesai1or** [comment]
Since the new coverage command is running the unit and integration tests, can't we remove the build steps that run the tests without the coverage report? No sense in running the same test suites twice for every PR

**Comment by luu-alex** [comment]
@spacesailor24 since coverage has a requirement that lines > 80%, coverage tests will fail in the first few packages. I believe until we get coverage for certain suites, then we can start combining the build steps. until then we should have them seperate.
I agree, there are alot of coverage tests suites perhaps I could just stick to one, (either geth or ganache) to reduce them.

**Review by Spacesai1or — APPROVED** [review]
There doesn't seem to be different coverage reports for Geth HTTP vs Ganache HTTP, but I would think there would be given that we have certain tests that only run if the provider is Geth/Ganache

Perhaps we should just keep this as is for now and remove a suite of tests when coverage for each package is above 80%, as you've mentioned

**Review by jdevcs — APPROVED** [review]

**Pull request merged**